transport_plugins/bled112/iotile_transport_bled112/bled112.py

```diff
@@ -569,21 +579,40 @@ def _process_scan_event(self, response):
 
             if data[22] == 0xFF and data[23] == 0xC0 and data[24] == 0x3:
                 self._v1_scan_count += 1
-                self._parse_v1_advertisement(rssi, string_address, data)
+                info = self._parse_v1_advertisement(rssi, string_address, data)
             elif data[3] == 27 and data[4] == 0x16 and data[5] == 0xdd and data[6] == 0xfd:
                 self._v2_scan_count += 1
-                self._parse_v2_advertisement(rssi, string_address, data)
+                info, reading_time, stream, reading, \
+                    broadcast_toggle, counter, broadcast_multiplex = \
+                    self._parse_v2_advertisement(rssi, string_address, data)
             else:
                 pass # This just means the advertisement was from a non-IOTile device
         elif packet_type == 4:
             self._v1_scan_response_count += 1
-            self._parse_v1_scan_response(string_address, data)
+            info, reading_time, stream, reading = \
+                self._parse_v1_scan_response(string_address, data)
+
+        if info:
+            drop_broadcast = self._check_update_seen_broadcast(
+                sender, reading_time, stream, reading,
+                broadcast_toggle, counter=counter, channel=broadcast_multiplex)
+
+            if not self._throttle_scans or not drop_broadcast:
+                self._trigger_callback('on_scan', self.id, info, self.ExpirationTime)
+
+            # If there is a valid reading on the advertising data, broadcast it
+            if (not self._throttle_broadcast or not drop_broadcast) and \
+                    stream and stream not in [0xFFFF, 0x7FFF]:
+                io_tile_reading = IOTileReading(reading_time, stream, reading, reading_time=datetime.datetime.utcnow())
+                report = BroadcastReport.FromReadings(info['uuid'], [io_tile_reading], reading_time)
+                self._trigger_callback('on_report', None, report)
+
 
     def _parse_v2_advertisement(self, rssi, sender, data):
         """ Parse the IOTile Specific advertisement packet"""
 
         if len(data) != 31:
-            return
+            return None, None, None, None
```

Comment:
Can we get some comments or record of the changes we made to this function (`_parse_v2_advertisement`) ? It's a pretty core functionality and I don't follow why the return type here is 4 nones, and the return further on in the function gives back 7 values (and doesn't do any callbacks). 

Why did the callbacks get moved to the `_process_scan_event` function?

Reply:
I moved handling of callback up to _process_scan_event to remove code duplication.  
In case you see more harm that good or the logic is confusing, I can revert this changes and copy  call back handling in every parsing function, but still it will be changes in core functionality.

Reply:
Got it, thanks. I understand that we're still changing functionality with the debug path, but it's crucial that we keep a record of what changed here (preferably in the commit or RELEASE comments) in case we do need to roll something back that we didn't catch. The refactor to group repeated code is fine, it just seemed to be a change that wasn't a required functionality at first glance so I needed to mention it.

Reply:
I added a notice to commit message, thank you for pointing this out.